app/features/syllabus_generator/tools.py

```diff
@@ -0,0 +1,178 @@
+from langchain_core.pydantic_v1 import BaseModel, Field
+from typing import List, Dict, Optional
+from app.services.logger import setup_logger
+from langchain_core.output_parsers import JsonOutputParser
+from langchain_google_genai import GoogleGenerativeAI
+from langchain_core.prompts import PromptTemplate
+from fastapi import HTTPException
+
+logger = setup_logger(__name__)
+
+class SyllabusRequestArgs(BaseModel):
+    subject_topic: str
+    grade_level: str
+    subject: str
+    course_description: str
+    course_objectives: str
+    required_materials: str
+    grading_policy: str
+    course_outline: str
+    class_policies: str
+    instructor_name: str
+    instructor_title: str
+    important_dates: Optional[str] = None
+    learning_outcomes: Optional[str] = None
+    class_schedule: Optional[str] = None
+    instructor_contact: Optional[str] = None
+    additional_customizations: Optional[str] = None
+
+    def to_dict(self) -> dict:
```

Comment:
Use the built-in Pydantic method, `model_dump` to return the model fields as a dict rather than a new `to_dict` method